e2e/helpers/http.ts

```diff
@@ -1,141 +1,107 @@
-import { createHmac } from 'crypto'
-import { fetchWithExponentialTimeout } from './srcFetch'
+import { Page, expect } from '@playwright/test'
+import { navigateToHomepageIfPageHasNotBeenLoaded } from './localStorage'
 
-export const BASE64_ENCODING = 'base64'
-export const DEFAULT_GETDELETE_HEADERS = {
-  Accept: '*/*',
-  'User-Agent': 'SynapseWebClient',
-}
-export const DEFAULT_POST_HEADERS = {
-  ...DEFAULT_GETDELETE_HEADERS,
-  Accept: 'application/json; charset=UTF-8',
-  'Content-Type': 'application/json; charset=UTF-8',
-}
-
-function generateDigitalSignature(
-  data: string,
-  base64EncodedSecretKey: string,
-) {
-  const hash = createHmac(
-    'sha1',
-    Buffer.from(base64EncodedSecretKey, BASE64_ENCODING),
-  )
-    .update(data)
-    .digest(BASE64_ENCODING)
-  return hash
-}
-
-function getDigitalSignature(
-  uri: string,
-  username: string,
-  base64EncodedSecretKey: string,
-) {
-  const timestamp = new Date().toISOString()
-  const signature = generateDigitalSignature(
-    `${username}${uri}${timestamp}`,
-    base64EncodedSecretKey,
-  )
-  return {
-    userId: username,
-    signatureTimestamp: timestamp,
-    signature: signature,
-  }
-}
-
-export function getUserIdFromJwt(token: string) {
-  const payload = JSON.parse(
-    Buffer.from(token.split('.')[1], BASE64_ENCODING).toString(),
-  )
-  return payload.sub
+export enum BackendDestinationEnum {
+  REPO_ENDPOINT,
+  PORTAL_ENDPOINT,
 }
 
-function updateHeaders(
-  headers: { [key: string]: string },
-  uri: string,
-  accessToken?: string,
-  userName?: string,
-  apiKey?: string,
-) {
-  return {
-    ...headers,
-    ...(accessToken && {
-      'Access-Control-Request-Headers': 'authorization',
-      Authorization: `Bearer ${accessToken}`,
-    }),
-    ...(apiKey && userName && getDigitalSignature(uri, userName, apiKey)),
-  }
+export async function waitForSrcEndpointConfig(page: Page) {
+  // window only available after page has initially loaded
+  await navigateToHomepageIfPageHasNotBeenLoaded(page)
+  // ensure that endpoint config is set,
+  // ...so API calls point to the correct stack
+  await expect(async () => {
+    const response = await page.evaluate('window.SRC.OVERRIDE_ENDPOINT_CONFIG')
+    expect(response).not.toBeUndefined()
+  }).toPass()
```

Comment:
Method moved from e2e/helpers/verification.ts, so it is colocated with other methods that rely on evaluating the page to access SRC